crates/polars-time/src/round.rs

```diff
@@ -44,3 +44,43 @@ impl PolarsRound for DateChunked {
             .into_date())
     }
 }
+
+#[cfg(feature = "dtype-duration")]
+impl PolarsRound for DurationChunked {
+    fn round(&self, every: Duration, offset: Duration, _tz: Option<&Tz>) -> PolarsResult<Self> {
+        if every.negative {
+            polars_bail!(ComputeError: "cannot round a Duration to a negative duration")
+        }
+        if !every.is_constant_duration() {
+            polars_bail!(InvalidOperation: "Cannot round a Duration series to a non-constant duration.");
+        }
+        if !offset.is_constant_duration() {
+            polars_bail!(InvalidOperation: "Cannot offset a Duration series by a non-constant duration.");
+        }
```

Comment:
`offset` has been deprecated for `round` in a separate PR - rather than adding complexity here for something which is now deprecated anyway, can we just raise if someone passes `offset` when rounding a duration?

This would be backwards-compatible as rounding durations wasn't supported yet